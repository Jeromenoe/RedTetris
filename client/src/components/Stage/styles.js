import styled from 'styled-components'

export const StyledStage = styled.div`
		display: grid;
		grid-template-rows: repeat(
			${props => props.height}, ${props => props.cellSize + 'px'});
		grid-template-columns: repeat(${props => props.width}, ${props => props.cellSize + 'px'});
		grid-gap: 1px ;
		border: 2px solid rgb(231, 76, 60) ;
		background: rgb( 189, 195, 199);
		/* background-color: #ecf0f1; */
		width: ${props => props.width * props.cellSize + props.width + 'px'};
		height: ${props => props.height * props.cellSize + props.height + 'px'};
		margin: 10px;
		padding: 1px;
		margin-right: 0px;
`