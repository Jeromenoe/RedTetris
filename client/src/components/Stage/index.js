import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { getStage } from '../../selectors/game';

import Cell from '../Cell/index';
import { StyledStage } from './styles';

const CELL_SIZE = 21;

const Stage = () => {
	useEffect(() => {
		if (document.getElementById("styledTetrisWrapper")) {
			if (document.getElementById("styledTetrisWrapper") !== document.activeElement) {
				document.getElementById("styledTetrisWrapper").focus();
			}
		}
	}, []);
	const stage = useSelector(getStage);
	return (
		<StyledStage width={stage[0].length} height={stage.length} cellSize={CELL_SIZE}>
			{stage.map(row => row.map((cell, x) => <Cell key={x} type={cell}/>))}
		</StyledStage>
	)
}

export default Stage;