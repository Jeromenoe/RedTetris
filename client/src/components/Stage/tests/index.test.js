import React from 'react';
import Stage from '../index';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import { Provider } from 'react-redux';


describe('Testing Stage Component', () => {
	test('Test with styledTetrisWrapper div', async () => {
		const div = document.createElement('div');
		div.setAttribute('id', 'styledTetrisWrapper');
		document.body.appendChild(div);
		const mockStore = configureStore();
		const initialState = {
			me: {
				stage: [
					[0, 0, 'I', 'J'],
					[0, 0, 'I', 'J'],
					[0, 0, 'I', 'J'],
					[0, 0, 'I', 'J']
				]
			}
		};
		const store = mockStore(initialState);
		const tree = renderer.create(<Provider store={store}><Stage /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
		document.body.removeChild(div);
	});

	test('Test without styledTetrisWrapper div', async () => {
		const mockStore = configureStore();
		const initialState = {
			me: {
				stage: [
					[0, 0, 'I', 'J'],
					[0, 0, 'I', 'J'],
					[0, 0, 'I', 'J'],
					[0, 0, 'I', 'J']
				]
			}
		};
		const store = mockStore(initialState);
		const tree = renderer.create(<Provider store={store}><Stage /></Provider>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});