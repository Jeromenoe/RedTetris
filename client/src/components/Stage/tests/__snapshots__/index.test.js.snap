// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Stage Component Test with styledTetrisWrapper div 1`] = `
<div
  className="sc-gtsrHT dSQuVR"
  height={4}
  width={4}
>
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
</div>
`;

exports[`Testing Stage Component Test without styledTetrisWrapper div 1`] = `
<div
  className="sc-gtsrHT dSQuVR"
  height={4}
  width={4}
>
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM cFlnuR"
    color="ecf0f1"
    type={0}
  />
  <div
    className="sc-bdnxRM eJWXiu"
    color="08a9a9"
    type="I"
  />
  <div
    className="sc-bdnxRM ffkpFI"
    color="2471a3"
    type="J"
  />
</div>
`;
